openstudiocore/src/model/ElectricEquipmentITEAirCooled.cpp

```diff
@@ -0,0 +1,560 @@
+    {
+      result.push_back(ScheduleTypeKey("ElectricEquipmentITEAirCooled","CPU Loading"));
+    }
+    return result;
+  }
+
+  bool ElectricEquipmentITEAirCooled_Impl::hardSize() {
+    boost::optional<Space> space = this->space();
+    if (!space) {
+      return false;
+    }
+
+    this->makeUnique();
+
+    ElectricEquipmentITEAirCooledDefinition electricEquipmentITEAirCooledDefinition = this->electricEquipmentITEAirCooledDefinition();
+    for (LifeCycleCost cost : electricEquipmentITEAirCooledDefinition.lifeCycleCosts()) {
+      cost.convertToCostPerEach();
+    }
+
+    boost::optional<double> wattsperUnit = electricEquipmentITEAirCooledDefinition.wattsperUnit();
+    if (wattsperUnit) {
+      return true;
+    }
+
+    boost::optional<double> wattsperZoneFloorArea = electricEquipmentITEAirCooledDefinition.wattsperZoneFloorArea();
+    if (wattsperZoneFloorArea) {
+      return electricEquipmentITEAirCooledDefinition.setWattsperUnit(*wattsperZoneFloorArea * space->floorArea());
+    }
+
+    return false;
+  }
+
+  bool ElectricEquipmentITEAirCooled_Impl::hardApplySchedules() {
+    bool result = true;
+
+    //optional in EnergyPlus
+    boost::optional<Schedule> designPowerInputSchedule = this->designPowerInputSchedule();
+    if (designPowerInputSchedule) {
+      this->setDesignPowerInputSchedule(*designPowerInputSchedule);
+    }
+
+    //optional in EnergyPlus
+    boost::optional<Schedule> cPULoadingSchedule = this->cPULoadingSchedule();
+    if (cPULoadingSchedule) {
+      this->setCPULoadingSchedule(*cPULoadingSchedule);
+    }
+
+    return true;
+  }
+
+  double ElectricEquipmentITEAirCooled_Impl::multiplier() const {
+    boost::optional<double> value = getDouble(OS_ElectricEquipment_ITE_AirCooledFields::Multiplier, true);
+    OS_ASSERT(value);
+    return value.get();
+  }
+
+  bool ElectricEquipmentITEAirCooled_Impl::isMultiplierDefaulted() const {
+    return isEmpty(OS_ElectricEquipment_ITE_AirCooledFields::Multiplier);
+  }
+
+  bool ElectricEquipmentITEAirCooled_Impl::isAbsolute() const {
+    ElectricEquipmentITEAirCooledDefinition definition = electricEquipmentITEAirCooledDefinition();
+    if (definition.wattsperUnit()) {
+      return true;
+    }
+    return false;
+  }
+
+  ElectricEquipmentITEAirCooledDefinition ElectricEquipmentITEAirCooled_Impl::electricEquipmentITEAirCooledDefinition() const {
+    return this->definition().cast<ElectricEquipmentITEAirCooledDefinition>();
+  }
+
+  boost::optional<Schedule> ElectricEquipmentITEAirCooled_Impl::designPowerInputSchedule() const {
+    boost::optional<Schedule> result = getObject<ModelObject>().getModelObjectTarget<Schedule>(OS_ElectricEquipment_ITE_AirCooledFields::DesignPowerInputScheduleName);
+    if (!result) {
+      // search upwards
+      OptionalSpace space = this->space();
+      OptionalSpaceType spaceType = this->spaceType();
+      if (space) {
+        result = space->getDefaultSchedule(DefaultScheduleType::DesignPowerInputScheduleName);
```

Comment:
Was this added to the DefaultScheduleType object?  I didn't see that change in the IDD.

Reply:
This schedule represents adding new servers over time right?  Seems kind of detailed....

Reply:
The default is a schedule that is always 1?

Reply:
Basically there are two layers of schedules originally designed for this object: first layer is the fraction of equipment that is running among all equipment (DesignPowerInputSchedule), second layer is the CPU load fraction of the running equipment (CPULoadingSchedule). Their default are both always 1.

I changed in DefaultScheduleSet.hpp [https://github.com/NREL/OpenStudio/pull/3343/files#diff-03f5b453c4044b5abceabd9f90492cf4](url)   Not sure if this is enough. It works for now.